server/src/main/java/org/elasticsearch/index/IndexSettings.java

```diff
@@ -986,6 +994,7 @@ public IndexSettings(final IndexMetadata indexMetadata, final Settings nodeSetti
         skipIgnoredSourceRead = scopedSettings.get(IgnoredSourceFieldMapper.SKIP_IGNORED_SOURCE_READ_SETTING);
         indexMappingSourceMode = scopedSettings.get(SourceFieldMapper.INDEX_MAPPER_SOURCE_MODE_SETTING);
         recoverySourceEnabled = RecoverySettings.INDICES_RECOVERY_SOURCE_ENABLED_SETTING.get(nodeSettings);
+        recoverySourceSyntheticEnabled = scopedSettings.get(RECOVERY_SOURCE_SYNTHETIC_ENABLED_SETTING);
```

Comment:
I [updated the validation](https://github.com/elastic/elasticsearch/pull/114618/commits/6b08b40d2dc4472bc7b92ccb0c15540e7f1130f0) so that the setting cannot be applied if synthetic mode is not enabled.